### PR TITLE
prevent scheduling >=30 years into the future

### PR DESCRIPTION
**Pull request by olucurious**
Nice one open-sourcing this btw @shalvah 👍🏾 

Here are the changes:
- added maximumReminderInterval and a conditional for it
- added a test for 1000 years and Jan 2100 😀 (The world would be a different place, very unlikely Twitter will still be a thing and we'd be super lucky to be alive) 

I'm not super sure you'd like the anonymous function for getting the future time but I think it's cleaner versus adding more figures, dealing with leap years etc; please let me know if there's a better alternative or if I need to change anything

### CODE REVIEW

File: tests/unit/service.test.js [hunk 1]
```diff
@@ -43,6 +43,15 @@ test("doesn't set for time in past", async () => {
     expect(parsingResult.failure).toBe("TIME_IN_PAST");
 });
 
+test("doesn't set for time in far future (>=30 years)", async () => {
+    parsingResult = await parseReminderTime(createMention({text: "@RemindMe_OfThis Jan 2100"}));
+    expect(parsingResult.failure).toBe("TIME_IN_FAR_FUTURE");
+
+    parsingResult = await parseReminderTime(createMention({text: "@RemindMe_OfThis in 1000 years"}));
+    console.log(parsingResult);
```
Comment (shalvah) [inline tests/unit/service.test.js:51]: You don't need to log this, btw. It was probably just a debugging thing.

File: src/factory.service.js [hunk 2]
```diff
@@ -26,7 +26,15 @@ const makeService = (cache, twitter, notifications) => {
             const reminderTime = mainResult.start.date();
             // Minimum of 3 minutes from now
             const minimumReminderInterval = (new Date(Date.now() + (3 * 60 * 1000)));
-            if (reminderTime >= minimumReminderInterval) {
+            // Maximum of 30 years from now
+            const maximumReminderInterval = (new Date((d => d.setFullYear(d.getFullYear() + 30))(new Date)));
```
Comment (shalvah) [inline src/factory.service.js:30]: Lol, you're right; I do not like this. That's way too many brackets for me. How about:

```suggestion
            const maximumReminderInterval = new Date((new Date).setFullYear(d.getFullYear() + 30));
```

**Review by shalvah — CHANGES_REQUESTED** [review]

File: src/factory.service.js (hunk 2 above)
Comment (olucurious) [reply src/factory.service.js:30]: Awesome, this is cleaner 👍🏾 will fix asap

Comment (olucurious) [reply src/factory.service.js:30]: Hmm, do you mean this
```suggestion
            const maximumReminderInterval = new Date((new Date).setFullYear((new Date).getFullYear() + 30));
```

**Review by shalvah — APPROVED** [review]

**Comment by shalvah** [comment]
Thank you! 👍

**Pull request merged**